src/osd/PrimaryLogPG.cc

```diff
@@ -3320,6 +3320,46 @@ void PrimaryLogPG::cancel_manifest_ops(bool requeue, vector<ceph_tid_t> *tids)
   }
 }
 
+int PrimaryLogPG::has_manifest_chunk(ObjectContextRef obc, std::string& fp_oid) 
+{
+  int cnt = 0;
+  // head
+  for (auto &p : obc->obs.oi.manifest.chunk_map) {
+    if (p.second.oid.oid.name == fp_oid) {
+      cnt++;
+    }
+  }
+  // snap
+  SnapSet& ss = obc->ssc->snapset;
+  for (vector<snapid_t>::const_iterator p = ss.clones.begin();
+      p != ss.clones.end();
+      ++p) {
+    object_ref_delta_t refs;
+    ObjectContextRef obc_l = nullptr;
+    ObjectContextRef obc_g = nullptr;
+    hobject_t clone_oid = obc->obs.oi.soid;
+    clone_oid.snap = *p;
+    ObjectContextRef clone_obc = get_object_context(clone_oid, false);
+    if (!clone_obc) {
+      break;
+    }
+    get_adjacent_clones(clone_obc, obc_l, obc_g);
+    clone_obc->obs.oi.manifest.calc_refs_to_inc_on_set(
+      obc_g ? &(obc_g->obs.oi.manifest) : nullptr ,
+      nullptr,
```

Comment:
Missing obc_l?

I'm not sure this is actually correct, probably you want to start at the oldest clone and work forward to head populating only obc_l.

Reply:
Does this provide the same output as our intention?
This commit counts all chunks in the head object and then checks the latest clone in reverse order.
(head -> snap[0] -> snap[1] in the below example)

 // foo snap[1]:      [HI] [HI]
 // foo snap[0]: [er] [ai] [SI]
 // foo head   : [er] [bi] [SI]

Reply:
Hmm, I suppose it does.